updatecli/updatecli.d/golang-alpine.yaml

```diff
@@ -0,0 +1,88 @@
+---
+name: "Bump Golang Alpine version"
+scms:
+  k3s:
+    kind: "github"
+    spec:
+      user: "{{ .github.user }}"
+      email: "{{ .github.email }}"
+      username: "{{ .github.username }}"
+      token: "{{ requiredEnv .github.token }}"
+      owner: "{{ .k3s.org }}"
+      repository: "{{ .k3s.repo }}"
+      branch: "{{ .k3s.branch }}"
+
+pullrequests:
+  github:
+    title: "Bump golang:alpine image version"
+    kind: "github"
+    scmid: "k3s"
+    spec:
+      automerge: false
+
+sources:
+  golang-docker-image:
+    name: "Check golang image version in DockerHub"
+    kind: "dockerimage"
+    spec:
+      image: "golang"
+      versionfilter:
+        kind: "semver"
+        regex: "*"
+  golang-alpine-docker-image:
+    name: "Check golang:alpine image version in DockerHub"
+    kind: "dockerimage"
+    sourceid: "golang-docker-image"
+    dependson:
+      - "golang-docker-image"
+    spec:
+      image: "golang"
+      versionfilter:
+        kind: "regex"
+        pattern: '{{ source `golang-docker-image` }}-alpine'
+
+conditions:
+  dockerfile-dapper:
+    name: "Check if 'ARG GOLANG' is set in Dockerfile.dapper"
+    kind: "dockerfile"
+    scmid: "k3s"
+    sourceid: "golang-alpine-docker-image"
+    spec:
+      file: "Dockerfile.dapper"
+      instruction:
+        keyword: "ARG"
+        matcher: "GOLANG"
+  dockerfile-test:
+    name: "Check if 'ARG GOLANG' is set in Dockerfile.test"
+    kind: "dockerfile"
+    scmid: "k3s"
+    sourceid: "golang-alpine-docker-image"
+    spec:
+      file: "Dockerfile.test"
+      instruction:
+        keyword: "ARG"
+        matcher: "GOLANG"
+  dockerfile-manifest:
+    name: "Check if 'ARG GOLANG' is set in Dockerfile.manifest"
+    kind: "dockerfile"
+    scmid: "k3s"
+    sourceid: "golang-alpine-docker-image"
+    spec:
+      file: "Dockerfile.manifest"
+      instruction:
+        keyword: "ARG"
+        matcher: "GOLANG"
+
+targets:
+  dockerfiles:
+    name: "Bump golang:alpine image version in Dockerfiles"
+    kind: "file"
+    scmid: "k3s"
+    disablesourceinput: true
+    spec:
+      files:
+        - "Dockerfile.dapper"
+        - "Dockerfile.test"
+        - "Dockerfile.manifest"
+      matchpattern: 'ARG GOLANG=golang:v?\d+\.\d+\.\d+(-alpine\S+)?'
+      replacepattern: 'ARG GOLANG=golang:{{ source "golang-alpine-docker-image" }}'
```

Comment:
This might not be a good candidate for automation at this time. We can't actually bump to whatever the latest is; we are specifically checking to confirm that we're building each release with the same golang version as upstream. See https://github.com/k3s-io/k3s/blob/master/scripts/validate#L27-L33

Reply:
@brandond thanks for catching this. Yes, we can definitely restrict the update only to the Alpine version.

Just for me to understand how the upstream check works and to try to propose an alternative:

- I see that K3s master branch points to [Go 1.19 in go.mod](https://github.com/k3s-io/k3s/blob/master/go.mod#L3). Is this always configured to match Go's minor version used in [K8s master branch](https://github.com/kubernetes/kubernetes/blob/master/build/dependencies.yaml#L91)?
- If yes, then I can configure the update pipeline to also check the upstream file and update Alpine's version accordingly. Would this make sense?
- Otherwise, the update pipeline can retrieve the Go version from our go.mod and point Alpine to `golang:1.19-alpineX.Y` instead.

WDYT?

Reply:
> If this code just kept the golang version the same, while bumping alpine, that would be fine.

Got it! I'll update the PR to only bump Alpine then. Thanks for the explanation.

Reply:
@brandond I changed the logic to only bump the Alpine version used in the Golang image. I had to do some workarounds, because we can't simply bump each image to the latest Alpine version. Suppose that there is a drift in the Golang images in the Dockerfiles, then we need to check if for that Golang version, the corresponding Alpine version is available. Like Alpine 3.17 is available for `1.19.3-alpine3.17` but not for `1.19.2-alpine3.17`. An example PR is in https://github.com/macedogm/test-k3s/pull/23.

Reply:
@brandond I did some further improvements and simplifications in the pipeline. Please let me know WDYT.